Fog.js

```diff
@@ -465,53 +465,74 @@ function check_token_visibility(){
 	}
 	else if(!window.NEXT_CHECK_TOKEN_VISIBILITY  || (window.NEXT_CHECK_TOKEN_VISIBILITY - Date.now() <  -1000)){
 		window.NEXT_CHECK_TOKEN_VISIBILITY = Date.now();
-		setTimeout(do_check_token_visibility(), 1);
+		do_check_token_visibility();
 		return;
 	}
 	else {
 		window.NEXT_CHECK_TOKEN_VISIBILITY = Date.now() + 1000;
-		setTimeout(do_check_token_visibility, 1000);
+		do_check_token_visibility;
 		return;
 	}
 }
 
-function do_check_token_visibility() {
+async function do_check_token_visibility() {
 	console.log("do_check_token_visibility");
 	let canvas = document.getElementById("fog_overlay");
 
 	if (canvas.style.diplay == "none")
 		return;
-	let ctx = canvas.getContext("2d");
+	let ctx = canvas.getContext("2d",  { willReadFrequently: true });
 	let canvas2 = document.getElementById("raycastingCanvas");
-	let ctx2 = canvas2.getContext("2d");
+	let ctx2 = canvas2.getContext("2d",  { willReadFrequently: true });
 
+	let promises = [];
 	for (let id in window.TOKEN_OBJECTS) {
+	if (id.includes(window.PLAYER_ID)) {
+		let auraSelectorId = $(".token[data-id='" + id + "']").attr("data-id").replaceAll("/", "");
+		let auraSelector = ".aura-element[id='aura_" + auraSelectorId + "']";
+		$(auraSelector).show();

```

Comment:
I'm not sure this needs to be here and we don't always want to show the players aura. I think this will just cause it to flash as it gets hidden below anyway.

Reply:
There is actually a bug with this below. 

```
	if(!window.TOKEN_OBJECTS[id].options.hideaura && id != playerTokenId)
				$(auraSelector).show();
				
```

should be 
```
	if(!window.TOKEN_OBJECTS[id].options.hideaura || id == playerTokenId)
				$(auraSelector).show();
				
```

Reply:
Yes I was, your solution is much better. Thank you